src/sentry/api/endpoints/team_details.py

```diff
@@ -74,8 +76,27 @@ def put(self, request: Request, team) -> Response:
         :param string name: the new name for the team.
         :param string slug: a new slug for the team.  It has to be unique
                             and available.
+        :param string orgRole: an organization role for the team. Only
+                               owners can set this value.
         :auth: required
         """
+        if request.data.get("orgRole"):
+            if team.idp_provisioned:
+                return Response(
+                    {
+                        "detail": "This team is managed through your organization's identity provider."
+                    },
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+
+            elif roles.get_top_dog() not in request.access.get_organization_roles():
```

Comment:
I think it's worth calling out here that we've made an explicit decision to allow only "top dog" to do this action, because it will simplify a lot of user flows. 

You should provide 1-2 examples of complicated flows. The idea is that someone looking at this in the future will have a hint that `roles.get_top_dog()` is a landmine and they'll be more deliberate when changing it.